qp_klp/process_amplicon_job.py

```diff
@@ -262,24 +265,65 @@ def process_amplicon(mapping_file_path, qclient, run_identifier, out_dir,
-        # populated. Generate pfp_list from walking the expected location.
-        pfp_list = []
-        for root, dirs, files in walk(join(pipeline.output_path,
-                                           'GenPrepFileJob', 'PrepFiles')):
-            for prep_file in files:
-                if prep_file.endswith('.tsv'):
-                    pfp_list.append(join(root, prep_file))
+        # list. Replace sample-names w/tube-ids in all relevant prep-files.
+        preps = list(chain.from_iterable(gpf_job.prep_file_paths.values()))
+        map_sample_names_to_tube_ids(preps, sn_tid_map_by_project)
 
-    map_sample_names_to_tube_ids(pfp_list, sn_tid_map_by_project)
+        # if seq_pro is run, prep_file_paths will be populated by run().
+        for study_id in gpf_job.prep_file_paths:
+            for prep_file_path in gpf_job.prep_file_paths[study_id]:
+                metadata = pd.read_csv(prep_file_path,
+                                       delimiter='\t',
+                                       index_col='sample_name').to_dict(
+                    'index')
+
+                # determine data_type based on target_gene column.
+                target_gene = metadata[list(metadata.keys())[0]]['target_gene']
+                for key in {'16S', '18S', 'ITS'}:
+                    if key in target_gene:
+                        data_type = key
+                        break
+
+                data = {'prep_info': dumps(metadata),
+                        'study': study_id,
+                        'data_type': data_type}
+
+                reply = qclient.post('/qiita_db/prep_template/', data=data)
+                prep_id = reply['prep']
+
+                touched_studies_prep_info[study_id].append(prep_id)
+    else:
+        # replace sample-names w/tube-ids in all relevant prep-files.
+        map_sample_names_to_tube_ids(join(pipeline.output_path,
+                                          'GenPrepFileJob', 'PrepFiles',
+                                          ('230224_M05314_0347_000000000-KVMH3'
+                                           '.ABTX_20230227_11052.1.tsv')),
+                                     sn_tid_map_by_project)
+
+        # assume testing conditions and assign preps to study 1.
+        metadata = {
+            'SKB8.640193': {'primer': 'GTGCCAGCMGCCGCGGTAA',
+                            'barcode': 'GTCCGCAAGTTA',
+                            'platform': 'Illumina',
+                            'instrument_model': 'Illumina MiSeq'},
+            'SKD8.640184': {'primer': 'GTGCCAGCMGCCGCGGTAA',
+                            'barcode': 'GTCCGCAAGTTA',
+                            'platform': 'Illumina',
+                            'instrument_model': 'Illumina MiSeq'}}
+
+        data = {'prep_info': dumps(metadata),
+                'study': '1',
+                'data_type': '16S'}
+
+        reply = qclient.post('/apitest/prep_template/', data=data)
```

Comment:
This should be `reply = qclient.post('/qiita_db/prep_template/', data=data)`, no?